### PR TITLE
RHOAIENG-9822: chore(Makefile): allow not pushing built images in Makefile and allow skipping building dependent images

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-9822

## Description

Follow-up to

* https://github.com/opendatahub-io/notebooks/pull/600#issuecomment-2202366809 

## How Has This Been Tested?

* https://github.com/jiridanek/notebooks/actions/runs/10200632903 (push)
* https://github.com/jiridanek/notebooks/actions/runs/10200837674?pr=6 (pr)

the failure in ^^^ is infra instability

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,10 +62,19 @@ endef
 #   ARG 1: Image tag name.
 #   ARG 2: Path of image context we want to build.
 #   ARG 3: Base image tag name (optional).
+#
+# BUILD_DEPENDENT_IMAGES: only build images that were explicitly given given as a goal on commandline
```
Comment (caponetto) [inline Makefile:66]:
```suggestion
# BUILD_DEPENDENT_IMAGES: only build images that were explicitly given as a goal on command line
```

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,10 @@ IMAGE_REGISTRY   ?= quay.io/opendatahub/workbench-images
 RELEASE	 		 ?= 2024a
 # additional user-specified caching parameters for $(CONTAINER_ENGINE) build
 CONTAINER_BUILD_CACHE_ARGS ?= --no-cache
+# whether to build all dependent images or just the one specified
+BUILD_DEPENDENT_IMAGES ?= yes
+# whether to push the images to a registry as they are built
+PUSH_IMAGES ?= yes
```
Comment (caponetto) [inline Makefile:8]: Maybe mention them in the README?

Comment (jiridanek) [reply Makefile:8]: sure, thanks

**Comment by jstourac** [comment]
/lgtm

Thanks, this is nice! :star2:

**Pull request merged**